### PR TITLE
[BugFix] Fix prefix-caching issue

### LINKED ISSUE #3510: [Bug][Qwen3-Omni]: Enabling prefix caching will cause requests to send output audio to hang.
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>

### Your code version

v0.20.0

### 🐛 Describe the bug

```
stages:
  - stage_id: 0
    max_num_batched_tokens: 32768
    max_num_seqs: 64
    gpu_memory_utilization: 0.9
    trust_remote_code: true
    enable_prefix_caching: true
    devices: "0"
    default_sampling_params:
      temperature: 0.4
      top_p: 0.9
      top_k: 1
      max_tokens: 2048
      seed: 42
      repetition_penalty: 1.05
```
Stage-0 enables prefix caching.

`vllm serve Qwen/Qwen3-Omni-30B-A3B-Instruct --omni --port 28889 --no-async-chunk `

```
python vllm_omni/examples/online_serving/openai_chat_completion_client_for_multimodal_generation.py --model Qwen/Qwen3-Omni-30B-A3B-Instruct \
    --query-type text  --port 28889
```
The request received no response.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by amy-why-3459**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
Fix prefix-caching issue
## Test Plan
`pytest -sv tests/e2e/online_serving/test_qwen3_omni_expansion.py -m "full_model" --run-level "full_model"`
## Test Result
```
../.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: 14 warnings
  /data/why/.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_text_to_audio_001[default]
  /data/why/.venv/lib/python3.12/site-packages/pydub/utils.py:14: DeprecationWarning: 'audioop' is deprecated and slated for removal in Python 3.13
    import audioop

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--- Running Summary
=========== 38 passed, 3 skipped, 17 warnings in 3238.05s (0:53:58) ============
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [ ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/model_executor/models/qwen3_omni/qwen3_omni.py [hunk 1]
```diff
@@ -670,8 +670,13 @@ def talker_preprocess(self, input_ids: torch.Tensor, input_embeds: torch.Tensor,
 
         span_len = input_ids.shape[0]
         update_dict: OmniPayload = {}
-        if span_len > 1:
-            # prefill
+        # Prefix caching can reduce a new request's remaining prefill span to a
+        # single token. Use the runner-provided phase flag instead of span_len.
+        is_prefill = bool(payload.get("_omni_is_prefill", span_len > 1))
```
Comment (alex-jw-brooks) [inline vllm_omni/model_executor/models/qwen3_omni/qwen3_omni.py:678]: Thanks @amy-why-3459! This looks reasonable to me.

Maybe we should add  some common utils to check for things like whether or not it's in prefill, since other models will probably do similar things (e.g., qwen tts [here](https://github.com/alex-jw-brooks/vllm-omni/blob/e277feacaf859c1aa3f2f7354d6fc396cf06ba5d/vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_talker.py#L577)). Or at least add it to the OmniPayload typed dict

**Review by alex-jw-brooks — APPROVED** [review]

**Review by hsliuustc0106 — CHANGES_REQUESTED** [review]
BLOCKING:

- Test Coverage — Test results section is empty. Please paste the actual output from pytest -sv tests/e2e/online_serving/test_qwen3_omni_expansion.py -m "full_model" --run-level "full_model".

- Reliability/Safety — The fix depends on _omni_is_prefill and _omni_num_computed_tokens payload fields, but there's no documentation or code comment explaining who populates these fields. This creates implicit coupling. Please add a comment explaining where these fields are set.

VERDICT: REQUEST_CHANGES

File: vllm_omni/model_executor/models/qwen3_omni/qwen3_omni.py (hunk 1 above)
Comment (amy-why-3459) [reply vllm_omni/model_executor/models/qwen3_omni/qwen3_omni.py:678]: > Thanks @amy-why-3459! This looks reasonable to me.
> 
> Maybe we should add some common utils to check for things like whether or not it's in prefill, since other models will probably do similar things (e.g., qwen tts [here](https://github.com/alex-jw-brooks/vllm-omni/blob/e277feacaf859c1aa3f2f7354d6fc396cf06ba5d/vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_talker.py#L577)). Or at least add it to the OmniPayload typed dict

Hi Alex, I've noticed that even after the fix in this PR, there's still a precision issue with the boundaries. For example, this test case: `pytest -sv tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_large_image_to_text_audio_001 -m "full_model" --run-level "full_model"`. The input length for this test case is 2063 (128 * block_size + 15). I hope we can discuss this issue in detail. Looking forward to your reply.

Comment (amy-why-3459) [reply vllm_omni/model_executor/models/qwen3_omni/qwen3_omni.py:678]: > > Thanks [@amy-why-3459](https://github.com/amy-why-3459)! This looks reasonable to me.
> > Maybe we should add some common utils to check for things like whether or not it's in prefill, since other models will probably do similar things (e.g., qwen tts [here](https://github.com/alex-jw-brooks/vllm-omni/blob/e277feacaf859c1aa3f2f7354d6fc396cf06ba5d/vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_talker.py#L577)). Or at least add it to the OmniPayload typed dict
> 
> Hi Alex, I've noticed that even after the fix in this PR, there's still a precision issue with the boundaries. For example, this test case: `pytest -sv tests/e2e/online_serving/test_qwen3_omni_expansion.py::test_large_image_to_text_audio_001 -m "full_model" --run-level "full_model"`. The input length for this test case is 2063 (128 * block_size + 15). I hope we can discuss this issue in detail. Looking forward to your reply.

After modifying slot_mapping, the accuracy returned to normal.

**Review by linyueqian — APPROVED** [review]
lgtm

**Review by Gaohan123 — APPROVED** [review]
LGTM. Thanks

**Comment by linyueqian** [comment]
fix ci please
